src/main/java/org/jabref/logic/ai/chathistory/BibDatabaseChatHistoryManager.java

```diff
@@ -66,42 +66,46 @@ public AiChatHistory getChatHistory(Path bibDatabasePath, String citationKey) {
             @Override
             public List<ChatMessage> getMessages() {
                 Map<Integer, ChatHistoryRecord> messages = getMap(bibDatabasePath, citationKey);
-                return IntStream.range(0, messages.size())
-                                .mapToObj(key -> messages.get(key).toLangchainMessage())
-                                .toList();
+
+                return messages
+                        .entrySet()
+                        .stream()
```

Comment:
```suggestion
                        // we need to check all keys, because upon deletion, there can be "holes" in the integer
                        .stream()
```